extensions/cache/src/main/java/org/radargun/stages/cache/test/CacheInvocations.java

```diff
@@ -599,4 +606,98 @@ public Operation txOperation() {
          return async ? ASYNC_TX : NATIVE_TX;
       }
    }
+
+   public static final class GetViaStream<K, V> implements Invocation<V> {
+      public static final Operation GET_NULL = StreamingOperations.GET.derive("Null");
+      public static final Operation TX = StreamingOperations.GET.derive("TX");
+      private final StreamingOperations.StreamingCache<K> cache;
+      private final K key;
+      private V value;
+
+      public GetViaStream(StreamingOperations.StreamingCache<K> cache, K key) {
+         this.cache = cache;
+         this.key = key;
+      }
+
+      @Override
+      public V invoke() {
+         InputStream in;
+         Integer result = 0;
+         if ((in = cache.getViaStream(key)) == null)
+            return null;
+
+         try {
+            int read = 0;
+            byte[] arr = new byte[100];
+            while ((read = in.read(arr)) != -1) {
+               result = result + read;
+            }
+            in.close();
+         } catch (IOException e) {
+            e.printStackTrace();
+         }
+
+         return value = (V) result;
+      }
+
+      @Override
+      public Operation operation() {
+         return value == null ? GET_NULL : BasicOperations.GET;
+      }
+
+      @Override
+      public Operation txOperation() {
+         return TX;
+      }
+   }
+
+   public static final class PutViaStream<K, V> implements Invocation<Void> {
+      public static final Operation TX = StreamingOperations.PUT.derive("TX");
+      private final StreamingOperations.StreamingCache<K> cache;
+      private final K key;
+      private final V value;
+
+      public PutViaStream(StreamingOperations.StreamingCache<K> cache, K key, V value) {
```

Comment:
byte[] was used before I decided to add DataStream (former RandomStream). I opted to leave it in as side option, but I can get rid of it.